packages/web-features-figuring/src/baseline/index.ts

```diff
@@ -0,0 +1,228 @@
+import assert from "node:assert";
+
+import { Temporal } from "@js-temporal/polyfill";
+
+import { feature } from "../browser-compat-data/feature";
+import { Release } from "../browser-compat-data/release";
+import { browsers, highReleases, lowReleases } from "./core-browser-set";
+import { support } from "./support";
+import { Browser } from "../browser-compat-data/browser";
+import { query } from "../browser-compat-data";
+
+interface FeatureSelector {
+  compatKeys: [string, ...string[]];
+  checkAncestors: boolean;
+}
+
+interface SupportStatus {
+  compatKey?: string;
+  baseline: "low" | "high" | false;
+  baseline_low_date: string | null;
+  baseline_high_date: string | null;
+  support: Map<Browser, Release | undefined>;
+  toJSON: () => string;
+}
+
+export function computeBaseline(
+  featureSelector: FeatureSelector,
+): SupportStatus {
+  const { compatKeys } = featureSelector;
+  const keys = featureSelector.checkAncestors
+    ? compatKeys.flatMap(withAncestors)
+    : compatKeys;
+
+  const statuses = keys.map(calculate);
+
+  const baseline = minStatus(statuses);
+
+  let baseline_low_date = null;
+  if (baseline !== false) {
+    baseline_low_date = latestDate(
+      statuses.map((status) => status.baseline_low_date),
+    );
+  }
+
+  let baseline_high_date = null;
+  if (baseline === "high") {
+    baseline_high_date = latestDate(
+      statuses.map((status) => status.baseline_high_date),
+    );
+  }
+
+  return {
+    baseline,
+    baseline_low_date,
+    baseline_high_date,
+    support: minSupport(statuses.map((status) => status.support)),
+    toJSON: function () {
+      return jsonify(this);
+    },
+  };
+}
+
+function calculate(compatKey: string): SupportStatus {
+  const f = feature(compatKey);
+  const s = support(f, browsers);
+
+  const allReleases = f.supportedBy();
```

Comment:
This should use the `only` option, so we don't let non-real values in non-Baseline browsers render the feature unresolvable.

See also: https://github.com/ddbeck/strict-browser-compat-data/pull/1